### PR TITLE
fix(files_versions): when resolving the storage of a file the full pa…

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…th has to be taken into account

## Description
files_versions has been integrated with the new storage based versioning IVersionedStorage
Until today this was only tested with objectstore versioning which is hooked up as primary storage.

When trying to access a versioned storage which is mounted as non-root this logic did not work out.

## Motivation and Context
Allow to use versioned storage in mounted storages.

## How Has This Been Tested?
- under development in files_spaces

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by phil-davis** [comment]
@DeepDiver1975 changelog needed?

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 changelog needed?

yes - and I want to add a test for this .... Monday ;-)

**Comment by DeepDiver1975** [comment]
>  and I want to add a test for this

not that easy .... bloody codebase ....

File: apps/files_versions/lib/Storage.php [hunk 1]
```diff
@@ -161,8 +161,9 @@ public static function store($filename) {
 			}
 
 			list($uid, $filename) = self::getUidAndFilename($filename);
+			$fullFileName = "/$uid/files$filename";
```
Comment (jvillafanez) [inline apps/files_versions/lib/Storage.php:164]: Couple of tiny details here, which could be solved with a quick code comment:
* It seems there is a "/" missing: `/$uid/files/$filename`. I guess the filename starts with "/" so it isn't needed.
* $filename could be misunderstood as just the filename, not the path. I assume it contains the whole path, such as `/folder1/folder2/filename`

**Review by jvillafanez — APPROVED** [review]
Just tiny details, but approving anyway.

File: apps/files_versions/lib/Storage.php (hunk 1 above)
Comment (DeepDiver1975) [reply apps/files_versions/lib/Storage.php:164]: Push permission granted ... feel free to add a comment ... THX

Comment (VicDeo) [inline apps/files_versions/lib/Storage.php:164]: isn't `$filename` a full file name before the reassignment in the previous line?

Comment (DeepDiver1975) [reply apps/files_versions/lib/Storage.php:164]: in a simplistic scenario: yes
but getUidAndFilename is also taking care of resolving shares to the owner

**Review by AlexAndBear — APPROVED** [review]

**Pull request merged**